javascript/encoders.mjs

```diff
@@ -0,0 +1,312 @@
+import { readFileSync } from 'fs';
+import { InferenceSession, Tensor } from 'onnxruntime-node';
+import { PreTrainedTokenizer } from '@xenova/transformers';
+import sharp from 'sharp';
+
+/**
+ * A processor for text data that prepares input for the text encoder model.
+ */
+class TextProcessor {
+
+    /**
+     * Constructs a new TextProcessor instance.
+     *
+     * @param {string} configPath - The path to the configuration file for the text encoder.
+     * @param {string} tokenizerPath - The path to the tokenizer configuration file.
+     */
+    constructor(configPath, tokenizerPath) {
+        this.configPath = configPath;
+        this.tokenizerPath = tokenizerPath;
+
+        this.maxSeqLen = 0;
+        this.padTokenIdx = 0;
+        this.tokenizer = null;
+    }
+
+    /**
+     * Initializes the TextProcessor by loading configurations and setting up the tokenizer.
+     */
+    async init() {
+        var config = JSON.parse(readFileSync(this.configPath, { encoding: 'utf8' }));
+        if (config.text_encoder !== undefined) {
+            config = config.text_encoder;
+        }
+
+        this.maxSeqLen = config.max_position_embeddings;
+        this.padTokenIdx = config.padding_idx;
+
+        const tokenizerConfig = JSON.parse(readFileSync(this.tokenizerPath, { encoding: 'utf8' }));
+        this.tokenizer = new PreTrainedTokenizer(tokenizerConfig, config);
+        this.tokenizer.model_max_length = this.maxSeqLen;
+        this.tokenizer.pad_token_id = this.padTokenIdx;
+    }
+
+    /**
+     * Processes a list of text strings into model-ready format, including padding and attention masks.
+     *
+     * @param {Array<string>} texts - An array of text strings to process.
+     * @return {Object} The processed texts as model input features.
+     */
+    async process(texts) {
+
+        const encoded = await this.tokenizer(texts, {
+            addSpecialTokens: true,
+            returnAttentionMask: true,
```

Comment:
```suggestion
            add_special_tokens: true,
```
See [here](https://github.com/xenova/transformers.js/blob/1a9964fb09b8f54fcbeac46dc6aae8d76795809d/src/tokenizers.js#L2628-L2641).

I've also removed `returnAttentionMask` since this is always returned.